lib/samples/apply_mosaic_rule_to_rasters/apply_mosaic_rule_to_rasters.dart

```diff
@@ -0,0 +1,200 @@
+  ),
+  lockRaster(
+    'Lock Raster',
+    'Displays only the selected rasters specified in [MosaicRule.lockRasterIds].',
+  );
+
+  const MosaicMethodEnum(this.value, this.description);
+  final String value;
+  final String description;
+}
+
+class ApplyMosaicRuleToRasters extends StatefulWidget {
+  const ApplyMosaicRuleToRasters({super.key});
+
+  @override
+  State<ApplyMosaicRuleToRasters> createState() =>
+      _ApplyMosaicRuleToRastersState();
+}
+
+class _ApplyMosaicRuleToRastersState extends State<ApplyMosaicRuleToRasters>
+    with SampleStateSupport {
+  // Create a controller for the map view.
+  final _mapViewController = ArcGISMapView.createController();
+  // A flag for when the map view is ready and controls can be used.
+  var _ready = false;
+  // If the mosaic options should be shown.
+  var _showMosaicOptions = false;
+  // Current selected mosaic method.
+  var _selectedMosaicMethod = MosaicMethodEnum.objectID;
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: SafeArea(
+        top: false,
+        left: false,
+        right: false,
+        child: Stack(
+          children: [
+            Column(
+              children: [
+                Expanded(
+                  // Add a map view to the widget tree and set a controller.
+                  child: ArcGISMapView(
+                    controllerProvider: () => _mapViewController,
+                    onMapViewReady: onMapViewReady,
+                  ),
+                ),
+              ],
+            ),
+            // Display a progress indicator and prevent interaction until state is ready.
+            LoadingIndicator(visible: !_ready),
+          ],
+        ),
+      ),
+      bottomSheet: _buildBottomSheet(),
+    );
+  }
+
+  Future<void> onMapViewReady() async {
+    // Create a map with a topographic basemap style.
+    final map = ArcGISMap.withBasemapStyle(BasemapStyle.arcGISTopographic);
+    _mapViewController.arcGISMap = map;
+    //Create a Raster with the provided URL
+    final raster = ImageServiceRaster(
+      uri: Uri.parse(
+        'https://sampleserver7.arcgisonline.com/server/rest/services/amberg_germany/ImageServer',
+      ),
+    );
+
+    //Create a RasterLayer with the Raster
+    final rasterLayer = RasterLayer.withRaster(raster);
+    await rasterLayer.load();
+    //Set a default MosaicRule to the RasterLayer
+    raster.mosaicRule = MosaicRule()..mosaicMethod = MosaicMethod.none;
+    //Add the RasterLayer to the operational layers of the map
+    map.operationalLayers.add(rasterLayer);
+    await _mapViewController.setViewpointCenter(
+      rasterLayer.fullExtent!.center,
+      scale: 25000,
+    );
+    // Set the ready state variable to true to enable the sample UI.
+    setState(() => _ready = true);
+  }
+
+  // Update the mosaic method of the raster layer.
+  Future<void> _updateMosaicMethod() async {
```

Comment:
This method doesn't use `await` anywhere, so it doesn't have to use the `async` keyword. And it can return simply `void`. And the calls to `setState(() => _ready = false)` and `setState(() => _ready = true)` won't actually do anything and are unnecessary.